### PR TITLE
Add quantile to dataframe groupby

### LINKED ISSUE #319: Implement `DataFrame.groupby().quantile()`
Logic Similar to #315 

Reference: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.DataFrameGroupBy.quantile.html

### PR DESCRIPTION
**Pull request by V1NAY8**
Closes #319 

Re used Most of the logic. 

@sethmlarson / @stevedodson Please review 

Ask jenkins to test this

### CODE REVIEW

**Comment by sethmlarson** [comment]
jenkins test this please

File: eland/operations.py [hunk 1]
```diff
@@ -1408,3 +1438,18 @@ def es_info(self, query_compiler, buf):
     def update_query(self, boolean_filter):
         task = BooleanFilterTask(boolean_filter)
         self._tasks.append(task)
+
+    @staticmethod
+    def quantile_to_percentile(quantile: Any) -> float:
```
Comment (sethmlarson) [inline eland/operations.py:1443]: I tend to dislike `staticmethods`, these are almost always better off as just a function.

Also since we're raising a `TypeError` for any non-`float`/`int` shouldn't `quantile` be `Union[float, int]`?

File: tests/dataframe/test_groupby_pytest.py [hunk 2]
```diff
@@ -230,3 +230,19 @@ def test_groupby_mode(self):
         match = "Currently mode is not supported for groupby"
         with pytest.raises(NotImplementedError, match=match):
             ed_flights.groupby("Cancelled").mode()
+
+    @pytest.mark.parametrize("dropna", [True, False])
+    @pytest.mark.parametrize("columns", ["Cancelled", ["dayOfWeek", "Cancelled"]])
+    @pytest.mark.parametrize("quantiles", [0.55, [0.2, 0.4, 0.6, 0.8]])
+    def test_groupby_aggs_quantile(self, dropna, columns, quantiles):
```
Comment (sethmlarson) [inline tests/dataframe/test_groupby_pytest.py:237]: Can we also add a test for an empty call like this: `.quantile()`?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
This is really fantastic :tada: Two comments for you:

File: eland/operations.py (hunk 1 above)
Comment (V1NAY8) [reply eland/operations.py]: Yep that's correct on typing, and agree on `staticmethods`

File: tests/dataframe/test_groupby_pytest.py (hunk 2 above)
Comment (V1NAY8) [reply tests/dataframe/test_groupby_pytest.py]: Yes :)

File: eland/operations.py [hunk 3]
```diff
@@ -1439,8 +1439,7 @@ def update_query(self, boolean_filter):
         task = BooleanFilterTask(boolean_filter)
         self._tasks.append(task)
 
-    @staticmethod
-    def quantile_to_percentile(quantile: Any) -> float:
+    def quantile_to_percentile(self, quantile: Union[int, float]) -> float:
```
Comment (sethmlarson) [inline eland/operations.py:1442]: Sorry for being confusing, didn't mean to change this into a method (ie a function that has `self`). I meant change it into a global function within the `eland/operations.py` file.

File: tests/dataframe/test_groupby_pytest.py [hunk 4]
```diff
@@ -232,16 +232,23 @@ def test_groupby_mode(self):
             ed_flights.groupby("Cancelled").mode()
 
     @pytest.mark.parametrize("dropna", [True, False])
+    @pytest.mark.parametrize(
+        ["func", "args"],
```
Comment (sethmlarson) [inline tests/dataframe/test_groupby_pytest.py:236]: :tada:

**Review by sethmlarson — CHANGES_REQUESTED** [review]
One comment then we can merge:

File: eland/operations.py (hunk 3 above)
Comment (V1NAY8) [reply eland/operations.py]: Yep got it. 💯

**Review by sethmlarson — APPROVED** [review]
LGTM

**Comment by sethmlarson** [comment]
jenkins test this please
